src/libs/auth.ts

```diff
@@ -0,0 +1,34 @@
+import { Request, Response, NextFunction } from 'express'
+import {
+  UnauthorizedError,
+  WrongTokenFormatError
+} from './errors'
+import { PrismaClient, User } from '@prisma/client'
+
+const prisma = new PrismaClient()
+
+/**
+ * Middleware to verify the webhook request is authenticated
+ *
+ */
+export async function authMiddleware(
+  req: Request,
+  res: Response,
+  next: NextFunction
+) {
+  if (req.headers?.authorization?.split(' ')[0] !== 'Bearer') {
```

Comment:
the webhook secret is not going to be an auth header. It will be in a header called `X-Webhook-Secret`

Reply:
it will also not be a bearer token, just a secret.

Reply:
addressed